kafka-0.10.x/src/main/scala/monix/kafka/KafkaProducer.scala

```diff
@@ -91,47 +87,51 @@ object KafkaProducer {
         val asyncCb = Callback.forked(cb)(s)
         val connection = StackedCancelable()
         // Forcing asynchronous boundary
-        sc.executeAsync(() =>
-          self.synchronized {
-            if (isCanceled) {
-              asyncCb.onSuccess(None)
-            } else {
-              val isActive = Atomic(true)
-              val cancelable = SingleAssignCancelable()
-              try {
-                // Force evaluation
-                val producer = producerRef
-
-                // Using asynchronous API
-                val future = producer.send(
-                  record,
-                  new KafkaCallback {
-                    def onCompletion(meta: RecordMetadata, exception: Exception): Unit =
-                      if (isActive.getAndSet(false)) {
-                        connection.pop()
-                        if (exception != null)
-                          asyncCb.onError(exception)
-                        else
-                          asyncCb.onSuccess(Option(meta))
-                      } else if (exception != null) {
-                        s.reportFailure(exception)
-                      }
+        sc.executeAsync(() => {
+          if (isCanceled.get()) {
+            asyncCb.onSuccess(None)
+          } else {
+            val isActive = Atomic(true)
+            val cancelable = SingleAssignCancelable()
```

Comment:
This `cancelable` is never added to `connection`, do you mind fixing it along the way? Just `connection.push(cancelable)` should do the trick

Reply:
👍